### PR TITLE
SAIC-691 Remove DHCP port on IP address conflict

### PR DESCRIPTION
**Pull request by antonf**

### CODE REVIEW

**Comment by roman-verchikov** [comment]
Looks good, +1

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.base.network
cloudferrylib/base/network.py:25: [E0711(notimplemented-raised), Network.get_func_mac_address] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:24: [W0613(unused-argument), Network.get_func_mac_address] Unused argument 'instance'
cloudferrylib/base/network.py:28: [E0711(notimplemented-raised), Network.create_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'mac'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'tenant_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'sg_ids'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'ip'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'net_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'keep_ip'
cloudferrylib/base/network.py:31: [E0711(notimplemented-raised), Network.delete_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:30: [W0613(unused-argument), Network.delete_port] Unused argument 'port_id'
cloudferrylib/base/network.py:34: [E0711(notimplemented-raised), Network.check_existing_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'network_id'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'ip_address'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'mac'
cloudferrylib/base/network.py:37: [E0711(notimplemented-raised), Network.get_security_groups] NotImplemented raised - should raise NotImplementedError
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1482: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1492: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1494: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1495: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1496: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1497: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1498: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1499: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:173: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:283: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 24)
cloudferrylib/os/network/neutron.py:763: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:867: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:892: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:923: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:984: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1375: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1382: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1386: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.base.network
cloudferrylib/base/network.py:25: [E0711(notimplemented-raised), Network.get_func_mac_address] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:24: [W0613(unused-argument), Network.get_func_mac_address] Unused argument 'instance'
cloudferrylib/base/network.py:28: [E0711(notimplemented-raised), Network.create_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'mac'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'tenant_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'sg_ids'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'ip'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'net_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'keep_ip'
cloudferrylib/base/network.py:31: [E0711(notimplemented-raised), Network.delete_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:30: [W0613(unused-argument), Network.delete_port] Unused argument 'port_id'
cloudferrylib/base/network.py:34: [E0711(notimplemented-raised), Network.check_existing_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'network_id'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'ip_address'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'mac'
cloudferrylib/base/network.py:37: [E0711(notimplemented-raised), Network.get_security_groups] NotImplemented raised - should raise NotImplementedError
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1489: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1499: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1501: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1502: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1503: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1504: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1505: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1506: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:173: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:285: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 24)
cloudferrylib/os/network/neutron.py:770: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:874: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:899: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:930: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:991: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1382: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1389: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1393: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.base.network
cloudferrylib/base/network.py:25: [E0711(notimplemented-raised), Network.get_func_mac_address] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:24: [W0613(unused-argument), Network.get_func_mac_address] Unused argument 'instance'
cloudferrylib/base/network.py:28: [E0711(notimplemented-raised), Network.create_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'mac'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'tenant_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'sg_ids'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'ip'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'net_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'keep_ip'
cloudferrylib/base/network.py:31: [E0711(notimplemented-raised), Network.delete_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:30: [W0613(unused-argument), Network.delete_port] Unused argument 'port_id'
cloudferrylib/base/network.py:34: [E0711(notimplemented-raised), Network.check_existing_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'network_id'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'ip_address'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'mac'
cloudferrylib/base/network.py:37: [E0711(notimplemented-raised), Network.get_security_groups] NotImplemented raised - should raise NotImplementedError
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1487: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1497: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1499: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1500: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1501: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1502: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1503: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1504: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:173: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:283: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 24)
cloudferrylib/os/network/neutron.py:768: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:872: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:897: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:928: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:989: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1380: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1387: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1391: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/network/neutron.py [hunk 1]
```diff
@@ -286,20 +286,29 @@ def get_network(self, network_info, tenant_id, keep_ip=False):
...
-                                                'mac_address', 'id']):
-            if (port['network_id'] == network_id) and (
-                    port['mac_address'] == mac):
+            networks = self.neutron_client.list_networks(
+                name=network_info['name'])['networks']
+            if len(networks) > 0:
+                return networks[0]
+        LOG.error('Failed to find network %s in tenant %s; keep_ip = %s',
+                  repr(network_info), tenant_id, keep_ip)
+        raise Exception("Can't find suitable network")
```
Comment (roman-verchikov) [inline cloudferrylib/os/network/neutron.py]: Let's use `AbortMigrationError` instead

**Comment by japaniel** [comment]
Passed in staging environment +1

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.base.network
cloudferrylib/base/network.py:25: [E0711(notimplemented-raised), Network.get_func_mac_address] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:24: [W0613(unused-argument), Network.get_func_mac_address] Unused argument 'instance'
cloudferrylib/base/network.py:28: [E0711(notimplemented-raised), Network.create_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'mac'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'tenant_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'sg_ids'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'ip'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'net_id'
cloudferrylib/base/network.py:27: [W0613(unused-argument), Network.create_port] Unused argument 'keep_ip'
cloudferrylib/base/network.py:31: [E0711(notimplemented-raised), Network.delete_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:30: [W0613(unused-argument), Network.delete_port] Unused argument 'port_id'
cloudferrylib/base/network.py:34: [E0711(notimplemented-raised), Network.check_existing_port] NotImplemented raised - should raise NotImplementedError
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'network_id'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'ip_address'
cloudferrylib/base/network.py:33: [W0613(unused-argument), Network.check_existing_port] Unused argument 'mac'
cloudferrylib/base/network.py:37: [E0711(notimplemented-raised), Network.get_security_groups] NotImplemented raised - should raise NotImplementedError
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1485: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1495: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1497: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1498: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1499: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1500: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1501: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1502: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:171: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:281: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 25)
cloudferrylib/os/network/neutron.py:766: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:870: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:895: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:926: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:987: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1378: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1385: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1389: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1, can be merged

**Pull request merged**